engine/src/main/java/ai/hhrdr/chainflow/engine/ethereum/InscriptionDataService.java

```diff
@@ -0,0 +1,75 @@
+package ai.hhrdr.chainflow.engine.ethereum;
+
+import ai.hhrdr.chainflow.engine.plugins.InscriptionHistoricalEventsHandler;
+import org.camunda.bpm.engine.impl.history.event.HistoryEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.web3j.crypto.Credentials;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.DefaultBlockParameterName;
+import org.web3j.protocol.core.methods.request.Transaction;
+import org.web3j.protocol.http.HttpService;
+import org.web3j.tx.RawTransactionManager;
+import org.web3j.tx.gas.DefaultGasProvider;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+
+
+@Service
+public class InscriptionDataService {
+
+    private final Web3j web3j;
+    private final Credentials credentials;
+
+    //TODO: Set a realistic gas limit based on prior knowledge or dynamic calculations
+    private final BigInteger gasLimit = BigInteger.valueOf(60000);
```

Comment:
Gas limit depends on the inscription size. If the inscription is really big, the transaction will fail due to out-of-gas error.